### DISCUSSION TITLE
Show image update(s) available

### DISCUSSION THREAD

**Discussion by alex3305** [louislam/dockge#239] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

API, UI Feature

### 🔖 Feature description

It would be really nice if Dockge would show whether a stack is out of date or whenever an update is available for images in a stack. Now I can click Update however much I want, but I don't know if it was necessary or not.

### ✔️ Solution

Show whether an update is available for a stack by periodically checking whether updates are available for images. This functionality would also overlap with #92, but is specific for a stack.

### ❓ Alternatives

Image management, as proposed by #92

**Comment by djismgaming** [comment]
Even a little colored icon beside the image tag could be great I guess:

Colors maybe like (simulated on screenshot):

- Green: Running the latest version of the image.
- Red: New image is available.
- Gray: Indicate an issue while checking if a new image is available.

These could be on a schedule, maybe. Showing on another color the "pending" status of the image check.

![image](https://github.com/louislam/dockge/assets/25346053/b51f73b3-e769-4852-95ba-124a50c31850)

**Comment by chall3ng3r** [comment]
This is much needed. Portainer have this via small dot in green or yellow which tells if there's update available.

**Comment by edrzns** [comment]
Agree, this is handy and actually needed functionality.
Other thing that could come handy would be notification implementation, that would cut off on using Diun or Watchtower or WUD for notification sending

**Comment by x1ao4** [comment]
This feature is nice.

**Comment by martin5233** [comment]
I would also like such a feature very much.

**Comment by GabrielPearce** [comment]
This would be an amazing feature I would love!

i am currently using dockcheck (https://github.com/mag37/dockcheck) which is clever and uses the diagest to determine if there is an update
the developer has a writeup about it here: https://mag37.org/posts/project_dockcheck/

**Comment by GIorfindel** [comment]
This is actually the thing preventing me from ditching portainer.

**Reply by djismgaming** [reply]
An idea for the time being. 

I set up Diun to notify me of updates via Ntfy. 

Then I go in Dockge and do the update.

**Reply by GIorfindel** [reply]
Yes, i saw watchtower and diun but I don't want to bother with setting up notifications or mails. Being able to see available updates directly in portainer is much more user friendly, so i will just stick with it for now.

**Comment by Hypnos-69** [comment]
Hello,

I've just discovered Dockge, which I've set up as a test. I really like its simplicity of use, which corresponds almost perfectly to my "basic" use of Docker.

Right now, I'm using Portainer BE because it's a simple way of alerting me to the availability of updates for my containers, via a green/orange badge system on each line of my stacks or containers. Even though Portainer is far too powerful for my basic use, this simple little feature makes all the difference for me.

As with Glorfindel, this tiny lack in Dockge means that I'm staying with Portainer BE in production for the moment. I'm going to continue to follow Dockge's development, and I think that as soon as this little feature (obviously requested by the community) will arrive, I will migrate from Portainer to Dockge because it's a really interesting solution.

Best regards,

Alex

**Comment by mikejaques** [comment]
This would be so nice.  It's how I quickly at a glance notice if the docker containers I have separately on my unRAID box need an update.

**Comment by tismofied** [comment]
Any Updates on this?

**Reply by di5cord20** [reply]
This feature would be great. I'm currently using docking-station for this purpose.

**Reply by tismofied** [reply]
Docking-station? [this one](https://github.com/LooLzzz/docking-station)

**Reply by di5cord20** [reply]
Yes, that's the docking-station I currently use.  It has a nice interface to show updates available and how mature those updates are.  I'd love that build into Dockge and not require a different app and interface to review.

**Reply by djismgaming** [reply]
Nice find! I do like the modularity of using a different application. I give it a try while this gets implemented in dockge.

Edit: It is glorious! Thanks for this! (love the direct links to the image's repos)

**Comment by deanfourie1** [comment]
While docking-station works, I would rather not spool up another container just to notify me of updates or I will have docker containers coming out my poop shoot.

Would be so much better to just keep it all in one,

Please integrate this feature request into DockGE.

**Comment by zandadoum** [comment]
i am trying to get away from portainer. loving dockge so far, specially now that i can add multiple server agents.
but i am really missing the update notification that portainer implemented last year

the suggestion [djismgaming](https://github.com/djismgaming) gave in january sounds great.

**Comment by theikid** [comment]
Any update on this? I'm using din with pushover but seing the status in dockage would definitively help. I don't want to use portainer :)

**Comment by SP3NGL3R** [comment]
This would be great. I'll look into bandaids (ex: diun) for now. I'm really trying to get away from Portainer too. This is my only 'would love ot have' feature that dockge doesn't have (yet). Thanks for a slick little app.

EDIT: DIUN->Gotify works really well for my use-case. I get an alert when a new image is available and bam! I'm easily updated via dockge. It took a few minutes to get it setup and I still haven't figured out how to not be an Admin everywhere in Gotify, but it works fine for the job. Until dockge adds this feature, DIUN->Gotify is fine.

**Comment by Generalnoob** [comment]
Adding my vote in to, I only use Dockge now it's perfect except for this one missing feature (to me)

**Comment by ghost** [comment]
Absolutely love it! Any update on this? Or may we discuss contributing to this feature, @louislam?

**Comment by tismofied** [comment]
I think @louislam is busy working on uptime-kuma 2.0 and as such he's not able to work on this feature request as of yet. Hopefully when he's done with that, he would  move to working on dockge 2.0 and have that feature built in 🙏

**Comment by breeegz** [comment]
I was updating all of my Dockge containers and had the same thought described in this Feature Request.  I find that I am clicking through all of my containers every Sunday, I've been burned by Watchtower in the past, so I don't want to make this an "auto-update" feature.

To add to the feature discussion, if Dockge could filter the containers displayed, with "update available" set as a quick filter then only containers needing an update are displayed, ready to be updated or check change logs for breaking changes.  I think the display filter could spawn out into it's own discussion, so I'll leave that subject alone.  I'm just thinking about the possibilities if that's the path that gets picked up.  (let me know if you want me to open that feature request and link back to this one)

**Comment by GameAreaCZ** [comment]
Yes please! I wanna switch to it from Portainer.

**Comment by WhiteSockedDancer** [comment]
After using watchtower for a long time, i switched to [WUD](https://github.com/getwud/wud). But there are also not many updates and i would love to have this feature in dockge.

WUD allows to specifiy some things in labels for each container. I use this for links to the change logs, better readability of containers (incl. icons). so i can easily check whats new before updating with a button press.

**Reply by evrial** [reply]
who cares?

**Reply by WhiteSockedDancer** [reply]
> who cares?

If we get this feature in dockge implemented having the option to add a path for the changelog would be quite nice. Maybe in this project are some good ideas how to implement this with some QoL-aspects.

**Reply by Ashkaan** [reply]
> who cares?

Everyone here.

If we can integrate our update process into our deployment tool, then most of us will have one less container to manager.

**Comment by Soljanka** [comment]
I'd love to have this feature aswell

**Comment by SonGokussj4** [comment]
THIS would be the best addition to now very functionally rich app. I was using some other means to know what containers are "to be updated" but here, I would just see what is 🔴 and click on `update` if I want.

Please, add this if it's even possible :-)
That would be so cool.

**Comment by hamphh** [comment]
I implemented this feature. Look at [https://github.com/louislam/dockge/discussions/844](https://github.com/louislam/dockge/discussions/844)
